documentation/hiera.md

```diff
@@ -0,0 +1,298 @@
+# Using Bolt with Hiera
+
+Because Bolt uses Puppet as a library, it has access to Hiera. Hiera is a
+built-in key-value configuration data lookup system, which lets you separate
+data from your code. There are some key differences in how Bolt and Puppet use
+Hiera, which this page covers in more detail.
+
+Before you start using Hiera in Bolt, get familiar with [how to use
+Hiera](https://puppet.com/docs/puppet/latest/hiera_intro.html).
+
+## Hiera configuration layers
+
+In Bolt, Hiera uses two independent layers of configuration when you look up
+data: the project layer and the module layer. This is different from Puppet,
+where Hiera has three independent layers of configuration. Bolt seraches the two
+layers of configuration in the order: project → module.
+
+### Project layer
+
+The configuration file for the project layer is located, by default, at
+`<PROJECT>/hiera.yaml`.
+
+The project layer is equivalent to the environment layer in Puppet. This is
+where you define most of your Hiera data hierarchy. Every project has a
+hierarchy configuration, which applies to all lookups made from that project.
+
+You can specify a different configuration file for the project layer by setting
+the `hiera-config` option in your project configuration, or by using the
+`--hiera-config` command-line option.
+
+📖  **Related information**
+
+- [bolt-project.yaml options](bolt_project_reference.md#hiera-config)
+- [*nix shell commands](bolt_command_reference.md)
+- [PowerShell cmdlets](bolt_cmdlet_reference.md)
+
+### Module layer
+
+The configuration for a module layer is located, by default, in the module's
+root directory at `<MODULE>/hiera.yaml`.
+
+The module layer sets default values and merge behavior for a module’s class
+parameters. The module layer comes last in Hiera’s lookup order, so environment
+data set by a user overrides the default data set by the module’s author.
+
+## Look up data from the command line
```

Comment:
```suggestion
## Look up data from the command-line
```
It actually seems like we use both pretty interchangeably throughout the docs, but at least being consistent within each page/section seems reasonable ;P

Reply:
We use 'command line' when referring to the command line itself, but hyphenate as 'command-line' when referring to the options.